api/pkg/transformer/feast/transformer.go

```diff
@@ -263,30 +372,39 @@ func (t *Transformer) buildEntitiesRequest(ctx context.Context, request []byte,
 	return entities, nil
 }
 
-func (t *Transformer) buildFeastFeatures(ctx context.Context, feastResponse *feast.OnlineFeaturesResponse, config *transformer.FeatureTable) (*FeastFeature, error) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "feast.buildFeastFeatures")
-	defer span.Finish()
+type cacheableFeatureData struct {
+	key   feast.Row
+	value FeatureData
+}
 
-	var columns []string
-	for _, entity := range config.Entities {
-		columns = append(columns, entity.Name)
-	}
-	for _, feature := range config.Features {
-		columns = append(columns, feature.Name)
-	}
+func (t *Transformer) buildFeastFeaturesData(ctx context.Context, feastResponse *feast.OnlineFeaturesResponse, config *transformer.FeatureTable, columns []string, entityIndexMap map[int]int) (FeaturesData, []cacheableFeatureData, error) {
```

Comment:
Can you make the return values from this method to only `([]EntityFeaturePair, error)`?
Current return values are redundant and complicate processing in the method.